state/backups/restore.go

```diff
@@ -125,33 +126,40 @@ func connectToDB(controllerTag names.ControllerTag, modelTag names.ModelTag, inf
 		pool *state.StatePool
 		err  error
 	)
-	const (
-		newStateConnDelay       = 15 * time.Second
-		newStateConnMinAttempts = 8
-	)
-	// TODO(katco): 2016-08-09: lp:1611427
-	attempt := utils.AttemptStrategy{Delay: newStateConnDelay, Min: newStateConnMinAttempts}
-
 	session, err := mongo.DialWithInfo(*info, mongoDefaultDialOpts())
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	defer session.Close()
 
-	for a := attempt.Start(); a.Next(); {
-		pool, err = state.OpenStatePool(state.OpenParams{
-			Clock:              clock.WallClock,
-			ControllerTag:      controllerTag,
-			ControllerModelTag: modelTag,
-			MongoSession:       session,
-			NewPolicy:          environsGetNewPolicyFunc(),
-		})
-		if err == nil {
-			return pool, nil
-		}
-		logger.Errorf("cannot open state, retrying: %v", err)
+	const (
+		newStateConnDelay    = 15 * time.Second
+		newStateConnAttempts = 8
+	)
+	err = retry.Call(retry.CallArgs{
```

Comment:
Can we split this out to a strategy pattern, that you've set up in the previous PRs.